nylon-threads/src/test/java/com/palantir/nylon/threads/NylonExecutorTest.java

```diff
@@ -132,4 +136,92 @@ void testThreadNamesAreUniqueWhenDelegateThreadsAreUnique() throws InterruptedEx
                     .isTrue();
         }
     }
+
+    @Test
+    void testFixedSizeExecutorAllowsQueueing() throws InterruptedException {
+        ExecutorService delegate = Executors.newCachedThreadPool();
+        try {
+            ExecutorService executor = NylonExecutor.builder()
+                    .name("foo")
+                    .executor(delegate)
+                    .maxThreads(1)
+                    .build();
+
+            CountDownLatch latch = new CountDownLatch(1);
+            int queuedTasks = 100;
+            CountDownLatch waitingLatch = new CountDownLatch(1);
+            AtomicInteger completed = new AtomicInteger();
+
+            executor.execute(() -> {
+                waitingLatch.countDown();
+                try {
+                    latch.await();
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+                completed.incrementAndGet();
+            });
+            // Subsequent tasks should be queued
+            for (int i = 0; i < 100; i++) {
+                executor.execute(completed::incrementAndGet);
+            }
+
+            waitingLatch.await();
+            assertThat(completed)
+                    .as("Tasks should all be queued behind the initial waiting task")
+                    .hasValue(0);
+            latch.countDown();
+            Awaitility.waitAtMost(Duration.ofSeconds(1))
+                    .untilAsserted(() -> assertThat(completed).hasValue(1 + queuedTasks));
+        } finally {
+            assertThat(MoreExecutors.shutdownAndAwaitTermination(delegate, Duration.ofSeconds(1)))
+                    .as("Delegate failed to stop")
+                    .isTrue();
+        }
+    }
```

Comment:
I like how these tests are quite readable despite the usual difficultly of testing concurrent things.

Reply:
Thanks :-)

I like to avoid abstracting out duplicate code when I test this sort of thing, I think it improves readability, and reduces the risk that future refactors will inadvertently invalidate existing tests.